src/basics.md

```diff
@@ -470,6 +471,48 @@ fn main() {
 }
 ```
 
+[ex-phone]: #ex-phone
+<a name="ex-phone"></a>
+## Extract phone numbers from text
+
+Attempts to process a user input as a US phone number.  Essentially, the
+expression is checking that there are 7 digits that come in groups of 3 3 4,
+removing any extra characters.  The phone number is then formatted for display.
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+extern crate regex;
+
+use std::io::stdin;
+use regex::Regex;
+#
+# error_chain!{
+#     foreign_links {
+#         Regex(regex::Error);
+#         Io(std::io::Error);
+#     }
+# }
+
+fn run() -> Result<()> {
+    println!("Input a phone number");
+    let mut phone_input = String::new();
+    // Regular expression is looking for 3 groups of digits, capturing them
+    let re = Regex::new(r".*([2-9]\d{2}).*(\d{3}).*(\d{4})")?;
```

Comment:
So I've investigated international phone numbers and their format is way different.  Here's the wikipedia article I'm investigating: https://en.wikipedia.org/wiki/National_conventions_for_writing_telephone_numbers
Because it would be easy to create an entire library out of the individual nation's formatting requirements I want to make this only function for US telephone numbers.  (While including the optional +1).  International conventions includes dots, dashes, spaces, and parenthesis optionally between digits.  I think +15551231234 and 5551231234 are both valid phone numbers.  Just that regex alone is actually a bit more complicated than the one you posted via mobile.  (and won't use the ^and$ because we are not validating we are extracting).

Anyway I'll throw something together that eliminates the false positives while looking good enough to reason (while only guaranteeing support for US).